model/transaction/event.go

```diff
@@ -70,6 +72,11 @@ type Event struct {
 	Experimental interface{}
 }
 
+type RUMV3Event struct {
```

Comment:
I think this is OK for the short term (expedient), but long term I don't think we should have a distinct "RUMV3Event" model type. This is a result of us treating the intake events as 1:1 with model objects. Instead, we should change processor/stream to stop assuming 1:1, and instead allow the decoders to produce >1 model objects.

Reply:
Not for now, but another way I've been thinking about going about this is something like:

- move all model types into the `model` package (i.e. collapse `model`, `model/transaction`, `model/span`, etc.)
- introduce a type like:

```go
package model

type Batch struct {
    Transaction	[]Transaction
    Span	[]Span
    Metricset	[]Metricset
    ...
}
```

The `DecodeFoo` methods would decode into a batch, by appending to the appropriate field(s).

Reply:
SGTM